src/main/objects/HarmonicFunction.js

```diff
@@ -90,7 +90,9 @@ function HarmonicFunction2(params){
             "revolution" : this.revolution.chordComponentString,
             "down" : this.down,
             "system" : this.system,
-            "mode" : this.mode
+            "mode" : this.mode,
+            "omit" : this.omit.slice(),
+            "extra" : this.extra.slice()
```

Comment:
Niestety to tak prosto nie zadziała... w args omit i extra to lista stringów, a tu wrzucasz liste ChordComponents. Będzie ten sam błąd co ostatnio, czyli chordComponent w chordComponent

Reply:
No właśnie miałem przykład, w którym to bez slice nie zadziałało, a ze slice już tak...

Reply:
Noo to dziwne... nie powinno zadziałać w żadnym wypadku. Trzebaby zrobić coś w stylu this.omit.map(function (cc) {return cc.chordChomponentString}) - z dokładnościa do składni "map" w js

Reply:
Sprawdziłem, składnia powyżej jest okej. Ponadto "map" w js tworzy nową listę także slice nie bedzię potrzebny. 
            "omit" : this.omit.map(function (cc) { return cc.chordComponentString; }),
            "extra" : this.extra.map(function (cc) { return cc.chordComponentString; })
Sprawdziłem u Ciebie i testy przechodzą, także lepiej bedzie tak od razu... wystarczy że przekleisz ;)